server/s3_put_multiobject_copy_action.cc

```diff
@@ -471,25 +474,53 @@ void S3PutMultipartCopyAction::copy_multipart_source_object() {
 
   total_objects = additional_object_metadata->get_number_of_fragments();
 
-  s3_log(S3_LOG_DEBUG, request_id,
-         "Total fragements/parts to be copied : (%d)\n", total_objects);
   extended_obj = additional_object_metadata->get_extended_object_metadata();
   const std::map<int, std::vector<struct s3_part_frag_context>>& ext_entries =
       extended_obj->get_raw_extended_entries();
-  for (int i = 0; i < total_objects; i++) {
+  for (unsigned int i = 0; i < total_objects; i++) {
     int ext_entry_index = 0;
-
     const struct s3_part_frag_context& frag_info =
         (ext_entries.at(i + 1)).at(ext_entry_index);
+    if (is_range_copy) {
+      struct S3ExtendedObjectInfo obj_info;
+      if (i == 0) {
+        // first part object
+        obj_info.start_offset_in_object = 0;
+      } else {
+        obj_info.start_offset_in_object =
+            extended_objects[i - 1].start_offset_in_object +
+            extended_objects[i - 1].object_size;
```

Comment:
Let's say we have a multipart object of size 10 with 3 parts: 0-3, 3-7, 7-10 and size 3,4,3 respectively. 
For i=0,  start_ofset is 0 (if condition)
For i=1, start_ofset is (start_ofset of object at i=0 which is 0 + size of object at i=0 which is 3) = 3
And so on . .

Reply:
Ok so for i=1, size of the object is that of 0th part.

>  we have a multipart object of size 10 with 3 parts: 0-3, 3-7, 7-10 and size 3,4,3 respectively.

small correct in ranges - multipart object of size 10 with 3 parts: **0-3, 4-7, 8-10**

Reply:
End is excluded in the range. It will be like [0-3), [3-7) and [7-10)
If we consider 1000 as obj size then offsets will be 0-299, 300-699, 700-999

Reply:
> End is excluded in the range. It will be like [0-3), [3-7) and [7-10)

thats like python range() function usage, we dont need to use that in explanation here. anyways i understand what you meant

> If we consider 1000 as obj size then offsets will be 0-299, 300-699, 700-999

this is proper